src/core/filters/packets/filter.rs

```diff
@@ -23,10 +39,79 @@ impl FilterPacket {
         let insns = program.get_instructions();
         let filter = BpfProg::try_from(unsafe { mem::transmute::<_, &[u8]>(insns) })?;
 
-        Ok(FilterPacket(eBpfProg::try_from(filter)?))
+        let ebpf_filter = eBpfProg::try_from(filter)?;
+        if ebpf_filter.len() > FILTER_MAX_INSNS {
+            bail!("Filter exceeds the maximum allowed size.");
+        }
+
+        Ok(FilterPacket(ebpf_filter))
     }
 
     pub(crate) fn to_bytes(&self) -> Result<Vec<u8>> {
         Ok(self.0.to_bytes())
     }
 }
+
+pub(crate) unsafe extern "C" fn prepare_load(
```

Comment:
The fn name is slightly misleading. It's following libbpf's callback naming (i.e: "when it's called") but not expressing a more useful "what it does".